src/Accelerators/Accelerator.cpp

```diff
@@ -13,23 +13,35 @@
 //===----------------------------------------------------------------------===//
 
 #include "src/Accelerators/Accelerator.hpp"
-#include <iostream>
-#include <vector>
+#include "src/Accelerators/NNPA/NNPAAccelerator.hpp"
 
 namespace onnx_mlir {
 namespace accel {
 
-std::vector<Accelerator *> *Accelerator::acceleratorTargets;
-
-Accelerator::Accelerator() {
-  if (acceleratorTargets == NULL)
-    acceleratorTargets = new std::vector<Accelerator *>();
+llvm::SmallPtrSet<Accelerator *, 2> Accelerator::accelerators;
+
+void Accelerator::create(Accelerator::Kind kind,
```

Comment:
all the work to create accelerators is done in this factory method. The caller just needs to pass the kind of accelerator that needs to be created and this code constructs it, initializes it and adds it to the list of accelerators.